apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -91,65 +93,105 @@ pub(super) fn validate_body_selection(
     field: &Component<FieldDefinition>,
     schema: &SchemaInfo,
     selection_node: &Node<Value>,
-) -> Result<(), Message> {
+) -> Vec<Message> {
     let coordinate =
         connect_directive_http_body_coordinate(&connect_directive.name, parent_type, &field.name);
 
-    let selection_str =
-        GraphQLString::new(selection_node, &schema.sources).map_err(|_| Message {
-            code: Code::GraphQLError,
-            message: format!("{coordinate} must be a string."),
-            locations: selection_node
-                .line_column_range(&schema.sources)
-                .into_iter()
-                .collect(),
-        })?;
-
-    let selection = JSONSelection::parse(selection_str.as_str()).map_err(|err| Message {
-        code: Code::InvalidJsonSelection,
-        message: format!("{coordinate} is not a valid JSONSelection: {err}"),
-        locations: selection_node
-            .line_column_range(&schema.sources)
-            .into_iter()
-            .collect(),
-    })?;
-
+    // Ensure that the body selection is a valid JSON selection string
+    let selection_str = match GraphQLString::new(selection_node, &schema.sources) {
+        Ok(selection_str) => selection_str,
+        Err(_) => {
+            return vec![Message {
+                code: Code::GraphQLError,
+                message: format!("{coordinate} must be a string."),
+                locations: selection_node
+                    .line_column_range(&schema.sources)
+                    .into_iter()
+                    .collect(),
+            }]
+        }
+    };
+    let selection = match JSONSelection::parse(selection_str.as_str()) {
+        Ok(selection) => selection,
+        Err(err) => {
+            return vec![Message {
+                code: Code::InvalidJsonSelection,
+                message: format!("{coordinate} is not a valid JSONSelection: {err}"),
```

Comment:
```suggestion
                message: format!("{coordinate} is not valid mapping syntax: {err}"),
```

Or something? Obviously this change isn't actually from this PR, but we should get rid of user-facing references to "JSONSelection" where we notice them.

Reply:
I guess that means we should also rename that code...

Reply:
This was already here, only the indent has changed in this PR - but it's a good point that we should sanitize anyplace that exposes "JSON Selection" to customers.

Reply:
I'll take it in hand to do that sanitization in a separate PR, as that'll fan out into a lot of snapshot updates.